relayer/chains/cosmos/provider.go

```diff
@@ -245,5 +247,7 @@ func (cc *CosmosProvider) BlockTime(ctx context.Context, height int64) (time.Tim
 }
 
 func (cc *CosmosProvider) SetMetrics(m *processor.PrometheusMetrics) {
```

Comment:
This is one of those kinds of fields that should never change after the struct is initialized. 

It would be better if the `PrometheusMetrics` value were part of the `NewCosmosProvider` function signature, then we can do away with the `SetMetrics` method altogether.

Reply:
Since `NewProvider` is an interface method that is called initially in `initConfig` before we ever construct the prometheus metrics, I kept the `SetMetrics` method but made sure it is only called once during  `StartRelayer` instead of during construction of the `CosmosChainProcessor`s, which allows the mutex to be isolated to the `TotalFees` as you suggested.